tests/integration.rs

```diff
@@ -3648,6 +3661,22 @@ mod tests {
                 .unwrap_or_default();
             aver_eq!(tb, reboot_count, 1);
 
+            // Resize of balloon will not available after reboot
+            aver!(tb, guest.get_total_memory().unwrap_or_default() > 982_000);
+
+            // Use balloon remove RAM from the VM
+            let desired_ram = 512;
+            resize_command(&api_socket, None, None, Some(desired_ram));
+
+            thread::sleep(std::time::Duration::new(10, 0));
+            aver!(tb, guest.get_total_memory().unwrap_or_default() > 491_000);
+            aver!(tb, guest.get_total_memory().unwrap_or_default() < 982_000);
+
+            // Use balloon add RAM to the VM
+            let desired_ram = 1024;
+            resize_command(&api_socket, None, None, Some(desired_ram));
+
+            thread::sleep(std::time::Duration::new(10, 0));
```

Comment:
I started investigating this issue and I found out the memory hotplug was triggering a guest reboot as you mentioned.
I have tested with QEMU and I got the same issue. I'm wondering if the virtio-balloon driver supports memory hotplug.

Reply:
QEMU and Cloud-Hypervisor both work. About Cloud-Hypervisor, the fix has been included in this PR.